### DISCUSSION TITLE
Levels of labels in address details

### DISCUSSION THREAD

**Discussion by molokanov50** [osm-search/Nominatim#2379] [Geocoding around the world]
Hi team,

According to the documentation, place output offers a set of labels which are used to describe address details. Here, https://nominatim.org/release-docs/develop/api/Output/#addressdetails, the labels are grouped by their administrative significance, so `state` and `region` appear to have the same administrative level.
In the same time, there are such address objects: https://nominatim.openstreetmap.org/search?q=novogireyevo&format=xml&addressdetails=1
We can see that `state` label actually corresponds to a region, and `region` label corresponds to a federal district, which, for Russia, is more generic than region. In other words, both labels of the same assumed level (from the viewpoint of the documentation) appeared simultaneously, and `region` is more generic than `state`. The same has place for `suburb` and `city_district`.
So are the labels somewhere described with some integer-value levels? Does the mapping change from country to country - maybe, for some countries, `region` can be, oppositely, more specific than `state`? Where can I find this mapping?

**Comment by lonvia** [comment]
The labels are derived from OpenStreetMap tagging practises and not from country-specific designations. They are assigned according to the place type and admin_level. See [this blog post](https://nominatim.org/2020/11/11/places-and-boundaries.html) for some insight.

If you need access to the integer [admin_levels as mapped in OSM](https://wiki.openstreetmap.org/wiki/Tag:boundary%3Dadministrative#10_admin_level_values_for_specific_countries), I recommend using the `geocodejson` format together with `addressdetails=1`. This returns an `admin` field with integer boundaries.

For example https://nominatim.openstreetmap.org/search?q=novogireyevo&format=geocodejson&addressdetails=1 has:
```
...
"admin": {
  "level3": "Central Federal District",
  "level4": "Moscow",
  "level8": "Novogireyevo District"
}
...
```
